src/aes_key.c

```diff
@@ -1099,3 +1205,16 @@ static int aes_key_blob_is_pkey_extractable(struct zpc_aes_key *aes_key, const u
 
 	return 0;
 }
+
+static void aes_key_add_ep11_header(struct zpc_aes_key *aes_key)
+{
+	struct ep11kblob_header *ep11hdr;
+
+	memset(aes_key->cur.sec, 0, sizeof(struct ep11kblob_header));
+	memmove(aes_key->cur.sec + sizeof(struct ep11kblob_header), aes_key->cur.sec, aes_key->cur.seclen);
+
+	ep11hdr = (struct ep11kblob_header *)aes_key->cur.sec;
+	ep11hdr->len = sizeof(struct ep11kblob_header) + aes_key->cur.seclen;
+	ep11hdr->version = TOKVER_EP11_AES_WITH_HEADER;
+	aes_key->cur.seclen = ep11hdr->len;
```

Comment:
You should zero the session field in the blob, now that you have added the header. Otherwise the session would specify a non-existing session id and it would probably fail.

Reply:
Fixed.